### PR TITLE
13 create posts feed page UI

### LINKED ISSUE #13: create Posts feed page UI

### PR DESCRIPTION
**Pull request by ayaali333**
close #13 
Screenshot (16)
Screenshot (17)

### CODE REVIEW

File: src/pages/post-feed/icos.tsx
Comment (MohammedAlabd) [inline src/pages/post-feed/icos.tsx:1]: I like that you put the icons in a separate file, good job

I would like to have them in a more generic place like the public folder

I want you to do some research on this, learn about static assets (because these are static assets) and see who to handle them and where to put them and let me know what you learn

Comment (MohammedAlabd) [reply src/pages/post-feed/icos.tsx:1]: you have a typo in the file name as well `icos` => `icons`

File: src/pages/post-feed/index.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import Image from 'next/image';
+import CreatPost from './CreatPost';
```
Comment (MohammedAlabd) [inline src/pages/post-feed/index.tsx:2]: this component is not available in this commit, maybe you forgot to commit it

it is fine to leave this for now, but next time make sure that each commit you do is executable

File: src/pages/post-feed/usersDummyData.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+export interface User {
+  id: number;
+  username: string;
+  userRank: string;
+  userPhoto: string;
+  text: string;
+  photo: string;
+}
+
+export const users: User[] = [
```
Comment (MohammedAlabd) [inline src/pages/post-feed/usersDummyData.ts:10]: I think you mean posts here and not users, right? 
the post is the main data we have, and the post will contain the user data in it

File: src/pages/post-feed/index.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              <div className="dropdown-end dropdown">
+                <div tabIndex={0} className="btn m-1">
+                  <svg
+                    xmlns="http://www.w3.org/2000/svg"
+                    fill="none"
+                    viewBox="0 0 24 24"
+                    className="inline-block h-5 w-5 stroke-current"
+                  >
+                    <path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M4 6h16M4 12h16M4 18h16" />
+                  </svg>
```
Comment (MohammedAlabd) [inline src/pages/post-feed/index.tsx]: ❌ 
put this SVG somewhere else and just imported in here

File: src/pages/post-feed/index.tsx [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                        </li>
+                      </ul>
+                    </details>
+                  </li>
+                  <li>
+                    <details>
+                      <summary>more</summary>
+                      <ul className="bg-base-100 p-2">
+                        <li>
+                          <a>Hide from active feed</a>
```
Comment (MohammedAlabd) [inline src/pages/post-feed/index.tsx]: ❌ 
the text `Hide from active feed` and all other text you have in the code should change to use internationalization, I know that we don't have it in the code right now, but after we add it you will have to change all of these

File: src/pages/post-feed/index.tsx [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                            <TwitterIcon />
+                          </a>
+                        </li>
+                        <li>
+                          <a>
+                            <InstagramIcon />
+                          </a>
+                        </li>
+                      </ul>
+                    </details>
```
Comment (MohammedAlabd) [inline src/pages/post-feed/index.tsx]: ❌ 
you could incapsulate all of this logic in one component

File: src/pages/post-feed/index.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+              height={500}
+              className="mask mask-squircle w-14"
+            />
+            <div className="flex flex-col">
+              <span className="ml-2">{user.username}</span>
+              <span className="ml-2 text-sm text-red-500">{user.userRank}</span>
+            </div>
+            <div className="navbar justify-end">
+              <div className="dropdown-end dropdown">
+                <div tabIndex={0} className="btn m-1">
```
Comment (MohammedAlabd) [inline src/pages/post-feed/index.tsx]: ❓ 
why `tabIndex={0}`

File: .eslintrc.json [hunk 7]
```diff
@@ -11,6 +11,18 @@
...
+      "error",
+      "ignorePackages",
+      {
+        "js": "never",
+        "jsx": "never",
+        "ts": "never",
+        "tsx": "never"
   }
+    ]
+}
```
Comment (MohammedAlabd) [inline .eslintrc.json]: ❓ 
can you explain to me why did you add each one of these?

File: next.config.js [hunk 8]
```diff
@@ -1,6 +1,9 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
+  images: {
+    domains: ['i.pinimg.com'],
+  },
```
Comment (MohammedAlabd) [inline next.config.js:5]: ❓ 
I didn't get why did you do that?

File: src/pages/post-feed/CreatPost.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-shadow */
+import Image from 'next/image';
+
+export default function CreatPost() {
+  interface User {
+    id: number;
+    userPhoto: string;
+  }
+
+  const user: User[] = [
```
Comment (MohammedAlabd) [inline src/pages/post-feed/CreatPost.tsx]: ❌ 
this variable is called `user` but it has an array, which is confusing, the name should change to be something else like `users` or `userList`

File: src/pages/post-feed/CreatPost.tsx [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+                />
+              </div>
+              <div className="px-24 text-center">
+                <div className=" flex flex-col px-24 align-top ">
+                  <p className="px-24 ">Lets GET STARTED WITH A QUESTION</p>
+                  <h3 className="px-24">What Would you Like to find in this community?</h3>
+                </div>
+              </div>
+              <div>
+                <div className="flex flex-col">
```
Comment (MohammedAlabd) [inline src/pages/post-feed/CreatPost.tsx]: ❓ 
in which case would the create post take a list of users?

**Review by MohammedAlabd — CHANGES_REQUESTED** [review]
your commit messages does not follow the [commitizen](https://github.com/commitizen/cz-cli) guid

File: src/pages/post-feed/icos.tsx
Comment (ayaali333) [reply src/pages/post-feed/icos.tsx:1]: Here is what I learned from assets "Static assets refer to files such as images, fonts, stylesheets, and other resources that are not dynamically generated by the server. These assets are meant to be served as-is, without any server-side processing. Next.js provides an efficient way to handle static assets and optimize their delivery for your web application"

File: src/pages/post-feed/index.tsx (hunk 1 above)
Comment (ayaali333) [reply src/pages/post-feed/index.tsx:2]: Did you mean here I shouldn't commit these two lines? 
if yes, can you explain to me why shouldn't? please

File: src/pages/post-feed/usersDummyData.ts (hunk 2 above)
Comment (ayaali333) [reply src/pages/post-feed/usersDummyData.ts:10]: Nope, I mean users cuz these data are related to users.

File: src/pages/post-feed/index.tsx (hunk 4 above)
Comment (ayaali333) [reply src/pages/post-feed/index.tsx]: Can you explain this to me, please?

File: src/pages/post-feed/index.tsx (hunk 5 above)
Comment (ayaali333) [reply src/pages/post-feed/index.tsx]: But all of them are just jsx, can I move them to another component and then import them into the index component?

File: src/pages/post-feed/CreatPost.tsx (hunk 10 above)
Comment (ayaali333) [reply src/pages/post-feed/CreatPost.tsx]: I didn't get your question.

File: next.config.js (hunk 8 above)
Comment (ayaali333) [reply next.config.js:5]: Ok, let me tell you what happen to me when I worked on images this error "Error: Invalid src prop [(https://i.pinimg.com/236x/da/c5/f8/dac5f861cf4a5d0cfbe8f310016bea28.jpg)](https://i.pinimg.com/236x/da/c5/f8/dac5f861cf4a5d0cfbe8f310016bea28.jpg) on `next/image`, hostname "i.pinimg.com" is not configured under images in your `next.config.js`
 This happened because I didn't add the domains of these images in the future we can replace this domain with the domains of users' images coming from the database. this is what I got from my search on this error.

File: .eslintrc.json (hunk 7 above)
Comment (ayaali333) [reply .eslintrc.json]: Ok......
first one cuz  I used tabIndex and why {tabIndex} to be honest I don't know because I copped this from daisyUI then this error appeared" should only be declared on interactive elements" Because of this I used this rule.

the second one wants to force me to use <### Link>" with "href" and I don't want it because if  I use it I can't put #<### a> inside  <### Link> because this is not permissible with the use of the last version of nextJs  


the third one was about missing file extension

File: src/pages/post-feed/index.tsx (hunk 1 above)
Comment (MohammedAlabd) [reply src/pages/post-feed/index.tsx:2]: because if we checkout to this commit and try to run the app it will crash, the error would be `cannot resolve CreatePost` or something like that because this component is not defined yet

File: src/pages/post-feed/usersDummyData.ts (hunk 2 above)
Comment (MohammedAlabd) [reply src/pages/post-feed/usersDummyData.ts:10]: yes they are related to users, but what you are showing her is posts, not users
the user is just the user data (username, image, email, ...etc) but the user will not have posts inside of it, otherwise, this would become very big, image facebook or Instagram, each user has thousands of posts, maybe even more, if we put the posts inside the user how big the user would be?

File: src/pages/post-feed/index.tsx (hunk 4 above)
Comment (MohammedAlabd) [reply src/pages/post-feed/index.tsx]: Imagine the case when we need to provide translation for the platform, how are we going to do it? 

this is a good task for you to do research on it, how to add translation to your website
I used i18n and also this is what we are using now for that, take a look at it and learn how it works

File: src/pages/post-feed/index.tsx (hunk 5 above)
Comment (MohammedAlabd) [reply src/pages/post-feed/index.tsx]: yes even if they are just JSX, you make a component out of them, these components are called dumb components, they have no logic, just JSX

this also reminds me of the idea of dumb vs smart components, this is a good topic to create some slides on 😉

File: .eslintrc.json (hunk 7 above)
Comment (MohammedAlabd) [reply .eslintrc.json]: hmmm, I am still thinks that you should not disable them from here on the whole project

each one of these are there for a reason, and I think we should respect these reasons, maybe you should try to fix the code in a way that doesn't complain on this

Comment (MohammedAlabd) [reply .eslintrc.json]: look up why they added this rules and what are we missing if we disable them

Comment (MohammedAlabd) [reply .eslintrc.json]: and for the third one, I didn't get the idea, what will happen if you remove this?

File: src/pages/post-feed/CreatPost.tsx (hunk 10 above)
Comment (MohammedAlabd) [reply src/pages/post-feed/CreatPost.tsx]: create post is a page where we go and create a post, can you explain why you are passing a list of users to this? 

can you explain to me how this logic is going to work? I didn't get it

File: public/feedPostIcons/Email.svg
Comment (MohammedAlabd) [inline public/feedPostIcons/Email.svg:1]: sorry, can you change these one more time? 
see Sara's PR and use the same file structure for that, please

`public/media/icons/[iconName].svg`

Comment (MohammedAlabd) [reply public/feedPostIcons/Email.svg:1]: do the same for the other icons as well

File: src/pages/post-feed/CreatPost.tsx [hunk 11]
```diff
@@ -2,12 +2,12 @@
 import Image from 'next/image';
 
 export default function CreatPost() {
-  interface User {
+  interface UserList {
```
Comment (MohammedAlabd) [inline src/pages/post-feed/CreatPost.tsx]: this name is misleading, it is referring to a list, yet the object is singular, so a name User was a better name for it

File: src/pages/post-feed/CreatPost.tsx [hunk 12]
```diff
@@ -2,12 +2,12 @@
...
     id: number;
     userPhoto: string;
   }
 
-  const user: User[] = [
+  const userList: UserList[] = [
```
Comment (MohammedAlabd) [inline src/pages/post-feed/CreatPost.tsx]: in this line, `const userList` is right, but the type `UserList` is a bad name

File: src/pages/post-feed/icons.tsx [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+export function TwitterIcon() {
+  return <Image src={Twittericon} alt="Facebook Icon" width={24} height={24} />;
+}
+
+export function EmailIcon() {
+  return <Image src={Email} alt="Facebook Icon" width={24} height={24} />;
+}
+export function SandwichButton() {
+  return (
+    <div tabIndex={0} className="btn m-1">
```
Comment (MohammedAlabd) [inline src/pages/post-feed/icons.tsx:25]: why this one is the only one that have a wrapper div and a tabIndex?

File: src/pages/post-feed/index.tsx (hunk 5 above)
Comment (ayaali333) [reply src/pages/post-feed/index.tsx]: Ok, I will do some slides about this😎

File: .eslintrc.json [hunk 14]
```diff
@@ -3,14 +3,25 @@
...
+    "import/extensions": [
+      "error",
+      "ignorePackages",
+      {
+        "js": "never",
+        "jsx": "never",
+        "ts": "never",
+        "tsx": "never"
+      }
+    ]
```
Comment (MohammedAlabd) [inline .eslintrc.json:25]: ❓ 
why do you still have this? just want to know what does it do?

File: src/pages/post-feed/icons.tsx [hunk 15]
```diff
@@ -1,9 +1,9 @@
...
-import facebook from '../../../public/feedPostIcons/facebook.svg';
-import Linkedinicon from '../../../public/feedPostIcons/LinkedInIcon.svg';
-import Twittericon from '../../../public/feedPostIcons/TwitterIcon.svg';
-import Email from '../../../public/feedPostIcons/Email.svg';
-import sandwichButton from '../../../public/feedPostIcons/SandwichButton.svg';
+import facebook from '../../../public/media/icons/facebook.svg';
+import Linkedinicon from '../../../public/media/icons/LinkedInIcon.svg';
+import Twittericon from '../../../public/media/icons/TwitterIcon.svg';
+import Email from '../../../public/media/icons/Email.svg';
+import sandwichbutton from '../../../public/media/icons/sandwichButton.svg';
```
Comment (MohammedAlabd) [inline src/pages/post-feed/icons.tsx:6]: ❌ 
you have changed the path, that is awesome

but you didn't move the files to the correct path, right?

File: src/pages/post-feed/CreatPost.tsx
Comment (MohammedAlabd) [inline src/pages/post-feed/CreatPost.tsx:1]: ❓ 
what does this component do?

File: src/pages/post-feed/Post.tsx
Comment (MohammedAlabd) [inline src/pages/post-feed/Post.tsx:1]: ❓ 
what does this component do?

File: src/pages/post-feed/icos.tsx
Comment (MohammedAlabd) [inline src/pages/post-feed/icos.tsx:1]: ❌ 
if you are not using this file, please remove it

**Review by MohammedAlabd — CHANGES_REQUESTED** [review]
please go through the old comments that I left for you and action them, because it seems like you forget some of them

File: .eslintrc.json (hunk 14 above)
Comment (ayaali333) [reply .eslintrc.json:25]: I forgot to answer this. Sorry, my bad.
**Missing file extension "ts"** When I import any component this error appears to me and when I searched for the reason I found that I am on the wrong path or **ts** is missing but I am on the right path also the extension not missing.

File: src/pages/post-feed/icons.tsx (hunk 15 above)
Comment (ayaali333) [reply src/pages/post-feed/icons.tsx:6]: Nope, I did, and everything worked correctly. Should I stage all SVGs again? 🤷‍♀️😁

File: src/pages/post-feed/CreatPost.tsx
Comment (ayaali333) [reply src/pages/post-feed/CreatPost.tsx:1]: This component has a responsibility to create a post "What I think about is there is a list of users it will coming from the database and when this user sign up we iterate around this list and appear this user  "If I am wrong please correct me. ✔

File: src/pages/post-feed/Post.tsx
Comment (ayaali333) [reply src/pages/post-feed/Post.tsx:1]: What I did here just moved the code from the main component to this one and then I imported this component inside the main one because I don't want to part the code I feel that is better. what do you think?

File: src/pages/post-feed/icos.tsx
Comment (ayaali333) [reply src/pages/post-feed/icos.tsx:1]: Here I import all the SVGs inside it and worked on image stuff and then  I imported just the icons names in one line inside the post component because I don't want any SVG import inside one component. Do you think this is wrong?

File: src/pages/post-feed/icons.tsx (hunk 15 above)
Comment (MohammedAlabd) [reply src/pages/post-feed/icons.tsx:6]: if this works then something is wrong, maybe you are not using this file at all, it could be something else

I order to figure this out, you need to answer the question, what does this file do?
are you even using it anywhere ?

File: src/pages/post-feed/CreatPost.tsx
Comment (MohammedAlabd) [reply src/pages/post-feed/CreatPost.tsx:1]: imagin a case when we have 500,000 user, do you think it is a good idea to search them in order to get the signed in user each time the user visits the page?

File: src/pages/post-feed/Post.tsx
Comment (MohammedAlabd) [reply src/pages/post-feed/Post.tsx:1]: yeah that is good

But I still have the same question, what does this component do?

File: src/pages/post-feed/icons.tsx (hunk 15 above)
Comment (ayaali333) [reply src/pages/post-feed/icons.tsx:6]: I answered this below.

File: src/pages/post-feed/icos.tsx
Comment (MohammedAlabd) [reply src/pages/post-feed/icos.tsx:1]: if you are using this component, why did you created the svg files in the public folder ?

File: src/pages/post-feed/Post.tsx
Comment (ayaali333) [reply src/pages/post-feed/Post.tsx:1]: Ok, this component has all the post stuff like the user photo and user name, text, images related to the post, and so on.

File: src/pages/post-feed/icos.tsx
Comment (ayaali333) [reply src/pages/post-feed/icos.tsx:1]: OK, look I import all SVGs in the first image below and then used them in the second and third images below. So what do you think should I remove the icons component and import all these SVGs inside the post component directly?🤔
Screenshot (18)
.
Screenshot (19)
Screenshot (20)

File: src/pages/post-feed/CreatPost.tsx
Comment (ayaali333) [reply src/pages/post-feed/CreatPost.tsx:1]: Aha, ok I will search for this.

File: src/pages/post-feed/Post.tsx
Comment (MohammedAlabd) [reply src/pages/post-feed/Post.tsx:1]: so will this render one post only? 
or a list of posts?

File: src/pages/post-feed/icos.tsx
Comment (MohammedAlabd) [reply src/pages/post-feed/icos.tsx:1]: according this code, you are not using this component `src/pages/post-feed/icos.tsx`, right?
